pkg/cloudprovider/aws/aws_routes.go

```diff
@@ -62,8 +62,13 @@ func (s *AWSCloud) ListRoutes(clusterName string) ([]*cloudprovider.Route, error
 			continue
 		}
 
+		instance, err := s.getInstanceById(instanceID)
+		if err != nil {
+			return nil, err
+		}
+		instanceName := instance.PrivateDNSName
 		routeName := clusterName + "-" + destinationCIDR
-		routes = append(routes, &cloudprovider.Route{routeName, instanceID, destinationCIDR})
+		routes = append(routes, &cloudprovider.Route{routeName, *instanceName, destinationCIDR})
```

Comment:
orEmpty(instanceName) removes the risk of dereferencing a null pointer (when you don't really want to handle the null case)